### PR TITLE
biguintcore.d: add return scope annotations

### PR DESCRIPTION
**Pull request by WalterBright**
These are the changes from https://github.com/dlang/phobos/pull/8076 made to std.internal.math.biguintcore to keep the PR size down. It adds `return` and `scope` annotations.

### CODE REVIEW

**Comment by WalterBright** [comment]
blocking https://github.com/dlang/phobos/pull/8104

**Review by atilaneves — APPROVED** [review]

**Comment by dkorpel** [comment]
Have you seen @nordlow 's WIP PR, https://github.com/dlang/phobos/pull/8081 , and have you seen that there's still three `@trusted` TODO's in here?

**Comment by WalterBright** [comment]
No, I didn't know #8081 existed. This one, however, passes and it will reduce the scope (!) of the remaining issues in #8081 so it can concentrate on getting it to pass.

I'm a big fan of incremental improvements!

**Comment by WalterBright** [comment]
Getting this merged will also enable https://github.com/dlang/phobos/pull/8104 which will further reduce the scope of the remaining work to be done.

**Comment by WalterBright** [comment]
Note that this PR is an exact copy of @nordlow 's work on biguintcore.d, it is not my work.

**Comment by dkorpel** [comment]
> Note that this PR is an exact copy of @nordlow 's work on biguintcore.d, it is not my work.

It's a copy of his WIP work, it's still being worked on in https://github.com/dlang/phobos/pull/8081

**Comment by dkorpel** [comment]
> No, I didn't know #8081 existed. This one, however, passes and it will reduce the scope (!) of the remaining issues in #8081 so it can concentrate on getting it to pass.

My concern is that when we rush this, we move from "memory corruption because of dmd bug" to "memory corruption because of overzealous `@trusted` in Phobos". A TODO comment can easily linger for years, it will only get fixed once an unfortunate programmer encounters memory corruption in his `@safe` BigInt code.

**Comment by nordlow** [comment]
> > No, I didn't know #8081 existed. This one, however, passes and it will reduce the scope (!) of the remaining issues in #8081 so it can concentrate on getting it to pass.
> 
> My concern is that when we rush this, we move from "memory corruption because of dmd bug" to "memory corruption because of overzealous `@trusted` in Phobos". A TODO comment can easily linger for years, it will only get fixed once an unfortunate programmer encounters memory corruption in his `@safe` BigInt code.

My intention is to remove them as soon as the dmd branch has been merged. Would it feel better If I made a list of them in them dmd in the dmd branch?

**Comment by WalterBright** [comment]
@nordlow I appreciate the work you're doing here. My goal here is to:

1. reduce the size of PRs
2. get the low-hanging fruit merged so it won't be delayed by work on the hard stuff
3. get the "leaf" modules working, so other modules that depend on them can be worked on in parallel.

(biguintcore appears to be a leaf file!)

I defer to you on this, @nordlow !

**Comment by dkorpel** [comment]
> My goal here is to:
>     1. reduce the size of PRs

@nordlow has already been doing that, see: https://github.com/dlang/phobos/pull/8076#issuecomment-842213613
GitHub shows referenced pull requests above that comment, most of them have been merged already!

**Comment by nordlow** [comment]
Please consider merging https://github.com/dlang/phobos/pull/8081 instead. I made to compile locally with any @trusted.

**Comment by WalterBright** [comment]
Closed in favor of #8081

**Pull request closed without merging**
